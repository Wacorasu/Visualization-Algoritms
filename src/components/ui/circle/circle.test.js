import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";

import { Circle } from "./circle.tsx";
import { ElementStates } from "../../../types/element-states.ts";

describe("Отрисовка компонента круг", () => {
  it("Круг рендерится без буквы без ошибок", () => {
    const tree = renderer
      .create(<Circle letter="" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг рендерится c буквами без ошибок", () => {
    const tree = renderer.create(<Circle letter="Тест" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг рендерится c head без ошибок", () => {
    const tree = renderer.create(<Circle letter="Тест" head='Тест'/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг рендерится  c параметром isSmall без ошибок", () => {
    const tree = renderer.create(<Circle letter="Тест" isSmall/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг рендерится  c index без ошибок", () => {
    const tree = renderer.create(<Circle letter="Тест" index={0}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг рендерится c react head без ошибок", () => {
    const circle = <Circle letter="Тест" isSmall/>
    const tree = renderer.create(<Circle letter="Тест" head={circle}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг рендерится c tail без ошибок", () => {
    const tree = renderer.create(<Circle letter="Тест" tail='Тест'/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг рендерится c react tail без ошибок", () => {
    const circle = <Circle letter="Тест"  isSmall/>
    const tree = renderer.create(<Circle letter="Тест" tail={circle}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it("Круг рендерится в состоянии modified без ошибок", () => {
    const tree = renderer.create(<Circle letter="Тест" state={ElementStates.Modified}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг рендерится в состоянии default без ошибок", () => {
    const tree = renderer.create(<Circle letter="Тест" state={ElementStates.Default}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг рендерится в состоянии changing без ошибок", () => {
    const tree = renderer.create(<Circle letter="Тест" state={ElementStates.Changing}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });


});
