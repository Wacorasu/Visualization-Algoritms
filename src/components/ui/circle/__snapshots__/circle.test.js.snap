// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Отрисовка компонента круг Круг рендерится  c index без ошибок 1`] = `
<div
  className="content "
  data-testid="circlesContainer"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circlesHead"
  />
  <div
    className="circle   default"
    data-testid="circleBorder"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Тест
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-testid="circlesTail"
  />
</div>
`;

exports[`Отрисовка компонента круг Круг рендерится  c параметром isSmall без ошибок 1`] = `
<div
  className="content "
  data-testid="circlesContainerHeadTail"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circlesHead"
  />
  <div
    className="circle  small default"
    data-testid="circleBorder"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Тест
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circlesTail"
  />
</div>
`;

exports[`Отрисовка компонента круг Круг рендерится c head без ошибок 1`] = `
<div
  className="content "
  data-testid="circlesContainer"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-testid="circlesHead"
  >
    Тест
  </div>
  <div
    className="circle   default"
    data-testid="circleBorder"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Тест
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circlesTail"
  />
</div>
`;

exports[`Отрисовка компонента круг Круг рендерится c react head без ошибок 1`] = `
<div
  className="content "
  data-testid="circlesContainer"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circlesHead"
  >
    <div
      className="content "
      data-testid="circlesContainerHeadTail"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="circlesHead"
      />
      <div
        className="circle  small default"
        data-testid="circleBorder"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          Тест
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="circlesTail"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-testid="circleBorder"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Тест
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circlesTail"
  />
</div>
`;

exports[`Отрисовка компонента круг Круг рендерится c react tail без ошибок 1`] = `
<div
  className="content "
  data-testid="circlesContainer"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circlesHead"
  />
  <div
    className="circle   default"
    data-testid="circleBorder"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Тест
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circlesTail"
  >
    <div
      className="content "
      data-testid="circlesContainerHeadTail"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="circlesHead"
      />
      <div
        className="circle  small default"
        data-testid="circleBorder"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          Тест
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="circlesTail"
      />
    </div>
  </div>
</div>
`;

exports[`Отрисовка компонента круг Круг рендерится c tail без ошибок 1`] = `
<div
  className="content "
  data-testid="circlesContainer"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circlesHead"
  />
  <div
    className="circle   default"
    data-testid="circleBorder"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Тест
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-testid="circlesTail"
  >
    Тест
  </div>
</div>
`;

exports[`Отрисовка компонента круг Круг рендерится c буквами без ошибок 1`] = `
<div
  className="content "
  data-testid="circlesContainer"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circlesHead"
  />
  <div
    className="circle   default"
    data-testid="circleBorder"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Тест
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circlesTail"
  />
</div>
`;

exports[`Отрисовка компонента круг Круг рендерится без буквы без ошибок 1`] = `
<div
  className="content "
  data-testid="circlesContainer"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circlesHead"
  />
  <div
    className="circle   default"
    data-testid="circleBorder"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circlesTail"
  />
</div>
`;

exports[`Отрисовка компонента круг Круг рендерится в состоянии changing без ошибок 1`] = `
<div
  className="content "
  data-testid="circlesContainer"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circlesHead"
  />
  <div
    className="circle   changing"
    data-testid="circleBorder"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Тест
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circlesTail"
  />
</div>
`;

exports[`Отрисовка компонента круг Круг рендерится в состоянии default без ошибок 1`] = `
<div
  className="content "
  data-testid="circlesContainer"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circlesHead"
  />
  <div
    className="circle   default"
    data-testid="circleBorder"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Тест
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circlesTail"
  />
</div>
`;

exports[`Отрисовка компонента круг Круг рендерится в состоянии modified без ошибок 1`] = `
<div
  className="content "
  data-testid="circlesContainer"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circlesHead"
  />
  <div
    className="circle   modified"
    data-testid="circleBorder"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Тест
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circlesTail"
  />
</div>
`;
